iib/workers/tasks/build_containerized_add.py

```diff
@@ -0,0 +1,376 @@
+            base_dir=temp_dir,
+            index_db=artifact_index_db_file,
+            bundles=resolved_bundles,
+            overwrite_csv=(prebuild_info['distribution_scope'] in ['dev', 'stage']),
+            graph_update_mode=graph_update_mode,
+        )
+
+        deprecation_bundles = get_bundles_from_deprecation_list(
+            present_bundles_pull_spec + resolved_bundles, deprecation_list or []
+        )
+
+        arches = prebuild_info['arches']
+        if deprecation_bundles:
+            deprecate_bundles_fbc_containerized(
+                bundles=deprecation_bundles,
+                base_dir=temp_dir,
+                binary_image=prebuild_info['binary_image'],
+                index_db_file=artifact_index_db_file,
+            )
+
+        os.makedirs(os.path.join(temp_dir, 'from_db'), exist_ok=True)
+        # get catalog from SQLite index.db (hidden db) - not opted in operators
+        catalog_from_db, _ = opm_migrate(
+            index_db=artifact_index_db_file,
+            base_dir=os.path.join(temp_dir, 'from_db'),
+            generate_cache=False,
+        )
+
+        catalog_from_index = localized_git_catalog_path
+
+        # we have to remove all `deprecation_bundles` from `catalog_from_index`
+        # before merging catalogs otherwise if catalog was deprecated and
+        # removed from `index.db` it stays on FBC (from_index)
+        # Therefore we have to remove the directory before merging
+        for deprecate_bundle_pull_spec in deprecation_bundles:
+            # remove deprecated operators from FBC stored in index image
+            deprecate_bundle = get_image_label(
+                deprecate_bundle_pull_spec, 'operators.operatorframework.io.bundle.package.v1'
+            )
+            bundle_from_index = os.path.join(catalog_from_index, deprecate_bundle)
+            if os.path.exists(bundle_from_index):
+                log.debug(
+                    "Removing deprecated bundle from catalog before merging: %s",
+                    deprecate_bundle,
+                )
+                shutil.rmtree(bundle_from_index)
+
+        # overwrite data in `catalog_from_index` by data from `catalog_from_db`
+        # this adds changes on not opted in operators to final
+        merge_catalogs_dirs(catalog_from_db, catalog_from_index)
+
+        # If the container-tool podman is used in the opm commands above, opm will create temporary
+        # files and directories without the write permission. This will cause the context manager
+        # to fail to delete these files. Adjust the file modes to avoid this error.
+        chmod_recursively(
+            temp_dir,
+            dir_mode=(stat.S_IRWXU | stat.S_IRWXG),
+            file_mode=(stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IWGRP),
+        )
+
+    # Write build metadata to a file to be added with the commit
+    set_request_state(request_id, 'in_progress', 'Writing build metadata')
+    write_build_metadata(
```

Comment:
It just struck me that we should consider writing `arches` to the build_metadata too. It might be helpful for use cases like v4.12 where we only want to build for 2 arches given OCP has stopped building for s390x and ppc64le

Reply:
Commit to change teh function signature : https://github.com/release-engineering/iib/pull/1274/commits/5d59cd78bb0796b5c3089e0b0ba5d6b51e426016